Services/Models/InternalDeviceState.cs

```diff
@@ -0,0 +1,247 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models
+{
+    public interface IInternalDeviceState
+    {
+        bool PropertyChanged { get; set; }
+        Dictionary<string, object> GetProperties();
+        bool HasProperty(string key);
+        object GetProperty(string key);
+        void SetProperty(string key, object value);
+        Dictionary<string, object> GetState();
+        void SetState(Dictionary<string, object> state);
+        bool HasStateValue(string key);
+        object GetStateValue(string key);
+        void SetStateValue(string key, object value);
```

Comment:
could you revisit these names consistency? you could also consider reusing one internal class for properties and state, e.g.
```
IInternalDeviceState->Properties->GetAll()
IInternalDeviceState->Properties->Set(k, v)
IInternalDeviceState->Properties->Get(k)
IInternalDeviceState->Properties->Has(k)
IInternalDeviceState->Properties->Changed(k)
IInternalDeviceState->State->GetAll()
IInternalDeviceState->State->Set(k, v)
IInternalDeviceState->State->Get(k)
IInternalDeviceState->State->Has(k)
IInternalDeviceState->State->Changed(k)
```